### ISSUE TITLE
ValueError thrown in trivial Keras model using tf.split

### ISSUE THREAD

**Issue by chrisbutner** [tensorflow/tensorflow#39838]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes, pasted below
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
Anaconda, "tensorflow-gpu"
- TensorFlow version (use command below):
2.1.0
- Python version:
3.7.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**
Keras model using tf.split (or tf.tile, or tf.zeros) hits the following exception, via "assert_shallow_structure" in nest.py, because one of the TensorSpecs is deeper in a sequence than the other (e.g. [[[TensorSpec]]] vs. [TensorSpec]). The actual trigger is tf.keras.models.load_model.

ValueError
Could not find matching function to call loaded from the SavedModel. Got:
  Positional arguments (1 total):
    * Tensor("inputs:0", shape=(None, 256, 8, 8), dtype=float32)
  Keyword arguments: {}

Expected these arguments to match one of the following 1 option(s):

Option 1:
  Positional arguments (1 total):
    * [TensorSpec(shape=(None, 256, 8, 8), dtype=tf.float32, name='inputs/0')]
  Keyword arguments: {}

**Describe the expected behavior**
Loading the SavedModel in Keras succeeds without an exception.

**Standalone code to reproduce the issue**
import tensorflow as tf
input = tf.keras.layers.Input(shape=(256, 8, 8), dtype="float32")
output, y = **tf.split(input, 2, axis=1)**
model = tf.keras.Model(input, output)
model.save("C:\\Users\\Public\\repro", save_format="tf")
tf.keras.models.load_model("C:\\Users\\Public\\repro")

**Workaround**
Wrapping the tf.split or other tf operation in a Keras Lambda layer works around the bug. Example code:
import tensorflow as tf
input = tf.keras.layers.Input(shape=(256, 8, 8), dtype="float32")
output, y = **tf.keras.layers.Lambda(lambda tensor : tf.split(tensor, 2, axis=1))(input)**
model = tf.keras.Model(input, output)
model.save("C:\\Users\\Public\\repro", save_format="tf")
tf.keras.models.load_model("C:\\Users\\Public\\repro")

**Other info / logs**

**Comment by chrisbutner** [comment]
The problem is that even if operations like tf.*, "output = input * 5", "output = input[:, :128, :, :]" are meant to be wrapped in a Keras layer (e.g. subclassed or Lambda), without doing so (a) the model will happily compile and train for hours, and (b) the exception thrown gives no indication of what the developer has done wrong. Fast-failing and clearer doc examples may both help at a project level.

**Comment by ravikyram** [comment]
@chrisbutner 

I am not seeing any issue with TF version 2.2.0.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/ed12871c438ae1ab8e3fbc0a0af35e31/untitled37.ipynb).Please verify once and close the issue. Thanks!

**Comment by chrisbutner** [comment]
Thanks. Confirmed fixed on 2.2, can train and predict after loading.
